app/src/main/java/ar/com/wolox/android/cookbook/twitterlogin/TwitterLoginRecipeFragment.kt

```diff
@@ -0,0 +1,111 @@
+package ar.com.wolox.android.cookbook.twitterlogin
+
+import android.content.Intent
+import android.net.Uri
+import android.view.View
+import android.widget.Toast
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.twitterlogin.model.YoutubeEmailResponse
+import ar.com.wolox.wolmo.core.fragment.WolmoFragment
+import com.twitter.sdk.android.core.Callback
+import com.twitter.sdk.android.core.TwitterSession
+import com.twitter.sdk.android.core.models.User
+import kotlinx.android.synthetic.main.fragment_twitter_login.*
+
+class TwitterLoginRecipeFragment : WolmoFragment<TwitterLoginRecipePresenter>(), TwitterLoginRecipeView {
+
+    override fun layout(): Int = R.layout.fragment_twitter_login
+
+    override fun init() {
+    }
+
+    override fun setListeners() {
+        vDefaultLoginBtn.setOnClickListener {
+            presenter.onTwitterLoginButtonClicked()
+        }
+
+        vCustomLoginBtn.setOnClickListener {
+            presenter.onLoginWithTwitterApiButtonClicked()
+        }
+
+        vGetProfileBtn.setOnClickListener {
+            presenter.onFetchDataButtonClicked()
+        }
+
+        vLogoutBtn.setOnClickListener {
+            presenter.onLogoutButtonClicked()
+        }
+    }
+
+    override fun showNetworkUnavailableError() {
+        Toast.makeText(context, getString(R.string.twitter_error_network_unavailable), Toast.LENGTH_LONG).show()
+    }
+
+    override fun setLoginCallback(callback: Callback<TwitterSession>) {
+        vDefaultLoginBtn.callback = callback
+    }
+
+    override fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?) {
+        super.onActivityResult(requestCode, resultCode, data)
+        vDefaultLoginBtn.onActivityResult(requestCode, resultCode, data)
+        presenter.onActivityResultFinished(requestCode, requestCode, data)
+    }
+
+    override fun showError(message: String) {
+        vDetails.text = message
+    }
+
+    override fun showAuthFail() {
```

Comment:
Maybe whe should change this.  I am thinking in something like showFailError( ErrorType ) and then the view decides what message is going to show